core/src/main/java/org/ehcache/EhcacheManager.java

```diff
@@ -242,10 +244,52 @@ private boolean isDiskTransient(Ehcache<?, ?> ehcache) {
     List<LifeCycled> lifeCycledList = new ArrayList<LifeCycled>();
 
     final Store.Provider storeProvider = serviceLocator.getService(Store.Provider.class);
-    Store.Configuration<K, V> storeConfiguration = new StoreConfigurationImpl<K, V>(config);
     if (config.getResourcePools().getResourceTypeSet().contains(ResourceType.Core.DISK)) {
-      storeConfiguration = new PersistentStoreConfigurationImpl<K, V>(storeConfiguration, alias);
+      LocalPersistenceService persistenceService = serviceLocator.getService(LocalPersistenceService.class);
+      
+      if (!config.getResourcePools().getPoolForResource(ResourceType.Core.DISK).isPersistent()) {
+        try {
+          persistenceService.destroyPersistenceSpace(alias);
+        } catch (CachePersistenceException cpex) {
+          throw new RuntimeException("Unable to clean-up persistence space for non-restartable cache " + alias, cpex);
+        }
+      }
+      try {
+        PersistenceSpaceIdentifier space = persistenceService.getOrCreatePersistenceSpace(alias);
+        serviceConfigs = Arrays.copyOf(serviceConfigs, serviceConfigs.length + 1);
+        serviceConfigs[serviceConfigs.length - 1] = space;
+      } catch (CachePersistenceException cpex) {
+        throw new RuntimeException("Unable to create persistence space for cache " + alias, cpex);
+      }
+    }
+    Serializer<K> keySerializer = null;
+    Serializer<V> valueSerializer = null;
+    SerializationProvider serialization = serviceLocator.getService(SerializationProvider.class);
+    if (serialization != null) {
+      try {
+        final Serializer<K> ks = serialization.createKeySerializer(keyType, config.getClassLoader(), serviceConfigs);
+        final Serializer<V> vs = serialization.createValueSerializer(valueType, config.getClassLoader(), serviceConfigs);
+        lifeCycledList.add(new LifeCycled() {
+
+          @Override
+          public void init() throws Exception {
+            //nothing
+          }
+
+          @Override
+          public void close() throws Exception {
+            ks.close();
+            vs.close();
+          }
+        });
+        keySerializer = ks;
+        valueSerializer = vs;
+      } catch (Throwable t) {
+        //TODO SerializationProvider doesn't require any exception behavior...
```

Comment:
What do you mean here?
So `create*Serializer` throws when no serialiser can be found, right? But if the problem happens with the value one, we may cause the key one to leak as it will never be closed.
Also, I assume code later on will break, right? Is there no way to improve on that? I know we can't fail fast and remain generic. But we could fail fast for the cases we know to be broken - like any off heap or disk resources and no serializers.